lib/spack/spack/package.py

```diff
@@ -705,11 +709,23 @@ def _expand_archive(stage, name=self.name):
                 placement = {'': placement}
             # Make the paths in the dictionary absolute and link
             for key, value in placement.iteritems():
-                link_path = join_path(self.stage.source_path, resource.destination, value)
+                target_path = join_path(self.stage.source_path, resource.destination)
+                link_path = join_path(target_path, value)
                 source_path = join_path(stage.source_path, key)
+
+                try:
+                    os.makedirs(target_path)
+                except OSError as err:
+                    if err.errno == errno.EEXIST and os.path.isdir(target_path):
+                        pass
+                    else: raise
+
+                # NOTE: a reasonable fix for the TODO above might be to have
+                # these expand in place, but expand_archive does not offer
+                # this
+
                 if not os.path.exists(link_path):
-                    # Create a symlink
-                    os.symlink(source_path, link_path)
+                    shutil.move(source_path, link_path)
```

Comment:
In this case, I preferred only having one copy of the result.  My real preference would be to just extract the package in place, but that seemed like it would take a deeper change than what I was planning for the other day.